eng/pipelines/libraries/variables.yml

```diff
@@ -5,3 +5,13 @@ variables:
     value: $(Build.SourcesDirectory)/src/libraries
   - name: pipelinesPath
     value: /eng/pipelines/libraries
+  - name: isOfficialBuild
```

Comment:
isn't this something that coreclr and installer would as well?

Reply:
It is unclear if it will be used I think

Reply:
don't you have any logic specific to internal builds?

Reply:
Please just be considerate with making additional fundamental changes at this point as it will reset all the testing and delay my live-live pipeline work. Frankly speaking I would be the happiest if we could merge this change in right now its current form and follow up with additional cleanups in separate PR's.

Reply:
I'm reviewing the PR right now and will comment on what points out. We can address those in a subsequent PR of course.

Reply:
@safern do we in libraries actually still need this variable? Should we instead just use the same pattern that other files use? https://github.com/dotnet/runtime/blob/LibrariesLiveLiveCi/eng/pipelines/common/checkout-job.yml#L11

Reply:
When going through the review I had similar opinions about isOfficialBuild; however, I think until we stand up the official build we will not know entirely how that will look. So making changes around it now seems like effort incorrectly spent.

Reply:
That was my plan. So at part of this iteration I wanted to just make the minimal set of changes on how we had set it up for libraries (using the same variables and all that). In a follow-up PR I plan working with @trylek and trying to define as much shared variables  as possible. 

But yes, this template can be shared and we can have a variable to know if it is internal or public build. As they use that to define the helix-queues. https://github.com/dotnet/runtime/blob/master/eng/pipelines/coreclr/templates/helix-queues-setup.yml#L118

Reply:
> So making changes around it now seems like effort incorrectly spent.

Yes, that's why I wanted to leave pretty much as is. Don't want to spin up CI again. 

Anyway I will have a follow-up PR probably to unify the test and build steps again for the time being as the machine provisioning delay is impacting the build time by a LOT.

Reply:
Thanks.

I believe code which is added via a PR should be reviewed carefully. In my experience, changing or getting rid of stuff after-merge is more costly/complex, mainly because of the lack of context. I would prefer having TODO comments where we intend to change the behavior later but I understand that we want to get this in asap therefore I'm fine with not adding them now.

Reply:
This applies to any kind of PR, not specific to this one.

Reply:
For this one since the changes were complicated and since we don’t really know what the final shape is going to be, I didn’t even knew which TODOs to add. After this is merged I will go and investigate what can be removed, honestly I submitted it as is as it was blocking @trylek. 

I think reviewing the build result and platform matrix should suffice.